tests/core/test_ensemble.py

```diff
@@ -1,566 +1,204 @@
     )
+    expected_events.update(predictions[expected_winner_idx].optional_events)
+    if last_action_was_action_listen:
+        expected_events.add(
+            DefinePrevUserUtteredFeaturization(
+                predictions[expected_winner_idx].is_end_to_end_prediction
+            )
+        )
 
-    tracker = DialogueStateTracker.from_events(
-        "rule triggering tracker",
-        evts=[
-            ActionExecuted(ACTION_LISTEN_NAME),
-            UserUttered(text="goodbye", intent={"name": "goodbye"}),
-        ],
-    )
-    policy_ensemble = unexpected_intent_policy_agent.policy_ensemble
-    prediction = policy_ensemble.probabilities_using_best_policy(
-        tracker, moodbot_domain, NaturalLanguageInterpreter()
+    # construct tracker
+    evts = (
+        [ActionExecuted(action_name=ACTION_LISTEN_NAME)]
+        if last_action_was_action_listen
+        else []
     )
+    tracker = DialogueStateTracker.from_events(sender_id="arbitrary", evts=evts)
 
-    test_utils.assert_predicted_action(prediction, moodbot_domain, "utter_goodbye")
-
-
-def test_ensemble_prevents_multiple_action_unlikely_intents(
-    monkeypatch: MonkeyPatch,
-    tmp_path: Path,
-    unexpected_intent_policy_agent: Agent,
-    moodbot_domain: Domain,
-):
-    monkeypatch.setattr(
-        UnexpecTEDIntentPolicy,
-        "predict_action_probabilities",
-        _action_unlikely_intent_for("greet"),
+    # get the best prediction!
+    best_prediction = default_ensemble.combine_predictions(
+        predictions, tracker, domain=domain
     )
 
-    tracker = DialogueStateTracker.from_events(
-        "rule triggering tracker",
-        evts=[
-            ActionExecuted(ACTION_LISTEN_NAME),
-            UserUttered(text="hello", intent={"name": "greet"}),
-            ActionExecuted(ACTION_UNLIKELY_INTENT_NAME),
-        ],
-    )
+    # compare events first ...
+    assert set(best_prediction.events) == expected_events
+    assert not best_prediction.optional_events
 
-    policy_ensemble = unexpected_intent_policy_agent.policy_ensemble
-    prediction = policy_ensemble.probabilities_using_best_policy(
-        tracker, moodbot_domain, NaturalLanguageInterpreter()
-    )
+    # ... then drop events and compare the rest
+    best_prediction.events = []
+    best_prediction.optional_events = []
+    predictions[expected_winner_idx].events = []
+    predictions[expected_winner_idx].optional_events = []
 
-    # prediction cannot be action_unlikely_intent for sure because
-    # the last event is not of type UserUttered and that's the
-    # first condition for `UnexpecTEDIntentPolicy` to make a prediction
-    assert (
-        moodbot_domain.action_names_or_texts[np.argmax(prediction.probabilities)]
-        != ACTION_UNLIKELY_INTENT_NAME
+    # ... not quite there yet, because old implementation creates a policy with
+    # best_policy.priority as priority and the first one is a tuple which then
+    # becomes a tuple with a tuple with an int, so...
```

Comment:
Can you clarify what you mean by "first one" here? I see in the code what happens, but the comment is confusing. Maybe replace the comment by a description of why the priority becomes a tuple upon `combine_predictions`.

Reply:
Actually, where does it become a tuple?

Reply:
Inside `PolicyPrediction` - the constructor puts that into a tuple of length 1 😅